plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log/action.py

```diff
@@ -50,6 +50,23 @@ def run(self, params={}):
         for log_entry in log_entries:
             log_entry["message"] = json.loads(log_entry.get("message", "{}"))
 
+            labels = log_entry.get("labels", [])
+            new_labels = []
+            if labels:
+                for label in labels:
+                    log_id = label.get("id")
+                    if log_id:
+                        response = self.connection.session.get(
+                            f"{self.connection.url}log_search/management/labels/{log_id}"
+                        )
+                        try:
+                            response.raise_for_status()
+                            new_labels.append(response.json().get("label", {}).get("name"))
+                        except Exception:
+                            continue
```

Comment:
This needs to be async'd to it's own function. This won't be very performant. 

broadcomm_symantic_endpoint_protection has a good example of how to do this: 

https://github.com/rapid7/insightconnect-plugins/blob/master/plugins/broadcom_symantec_endpoint_protection/icon_broadcom_symantec_endpoint_protection/util/api.py#L210